Bio/PDB/PDBList.py

```diff
@@ -35,18 +35,46 @@
 
 """Access the PDB over the internet (e.g. to download structures)."""
 
+from __future__ import annotations
 
+import collections
 import contextlib
+import dataclasses
 import ftplib
+import functools
 import gzip
 import os
 import re
 import shutil
+import socket
 import sys
+import time
 
 from urllib.request import urlopen
 from urllib.request import urlretrieve
 from urllib.request import urlcleanup
+from urllib.parse import urljoin
+from urllib.parse import urlunsplit
+
+from Bio.PDB.PDBExceptions import PDBException
+
+
+@dataclasses.dataclass
+class PDBServer:
+    """Protein Data Server class.
+
+    In charge of the interactions with the server.
+    """
+
+    domain: str
```

Comment:
I'd give these more descriptive names. Not sure if these are better but `base_url` and `pdb_path` seem a little better to me.

Reply:
`domain_url` and `pdb_dir_path` look like a good compromise!

Reply:
Sorry to insist, but the domain is a subpart of the URL. I do not feel comfortable to do this confusion on purpose (cf. https://en.wikipedia.org/wiki/URL)

And why not `pdb_dir_path`, if you think we gain in readability

Reply:
OK, let's keep `domain` and `pdb_dir_path`, compromise :)

Reply:
6baa32e22caa0b4f38be7df76df7fea5111b0ad8 :rocket: